### PR TITLE
Hydrate contact and facility in API userSettings

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Hydrate user's contact and facility when getting user settings. 
Use all subjects from user's contact and facility when checking if a report is sensitive. 

medic/cht-core#6660

# Code review checklist

- [x] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [x] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [x] Tested: Unit and/or e2e where appropriate
- [x] Internationalised: All user facing text
- [x] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Please review, @latin-panda . Thanks!

**Review by latin-panda — CHANGES_REQUESTED** [review]
Nice job! I left a couple of minor suggestions in line :)

File: api/tests/mocha/auth.spec.js [hunk 1]
```diff
@@ -303,5 +475,32 @@ describe('Auth', () => {
...
+        name: 'my-user-edited',
+        roles: [ '_admin' ],
+        type: 'user-settings',
+        some: 'field',
+        contact_id: 'my-user-contact',
+        facility_id: 'otherVille'
+      });
+      sinon.stub(db.medic, 'allDocs').rejects({ error: 'boom' });
+
+      return auth.getUserSettings({ name: 'steve' }).catch(err => {
```
[the remark this answers is not in the document]
Comment (dianabarsan) [reply api/tests/mocha/auth.spec.js:500]: yes! good catch!

**Review by latin-panda — APPROVED** [review]
Nice Job!✨✨

**Pull request merged**